source/extensions/filters/http/cache/cache_headers_utils.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include "envoy/common/time.h"
+#include "envoy/http/header_map.h"
+
+#include "absl/strings/string_view.h"
+#include "absl/time/time.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+using OptionalDuration = absl::optional<SystemTime::duration>;
+
+struct RequestCacheControl {
+  RequestCacheControl() = default;
+  explicit RequestCacheControl(absl::string_view cache_control_header);
+
+  // must_validate is true if 'no-cache' directive is present
+  bool must_validate_ = false;
+  bool no_store_ = false;
+  // 'no-transform' directive is not used now
+  bool no_transform_ = false;
+  // 'only-if-cached' directive is not used now
+  bool only_if_cached_ = false;
+  OptionalDuration max_age_;
+  OptionalDuration min_fresh_;
+  OptionalDuration max_stale_;
+};
+
+struct ResponseCacheControl {
+  ResponseCacheControl() = default;
+  explicit ResponseCacheControl(absl::string_view cache_control_header);
+
+  // must_validate is true if 'no-cache' directive is present; arguments are ignored for now
+  bool must_validate_ = false;
+  // no_store is true if any of 'no-store' or 'private' directives is present.
+  // 'private' arguments are ignored for now so it is equivalent to 'no-store'
+  bool no_store_ = false;
+  // 'no-transform' directive is not used now
+  bool no_transform_ = false;
+  // no_stale is true if any of 'must-revalidate' or 'proxy-revalidate' directives is present
+  bool no_stale_ = false;
+  // 'public' directive is not used now
+  bool is_public_ = false;
+  // max_age is set if to 's-maxage' if present, if not it is set to 'max-age' if present.
```

Comment:
Can you summarize in comments what each field means semantically, it's a pain to keep looking at the RFC.